keps/sig-storage/2485-read-write-once-pod-pv-access-mode/README.md

```diff
@@ -0,0 +1,972 @@
+alternative is to allow these volumes to be treated as ReadWriteOnce, however
+that would violate the intent of the user and so it is not recommended.
+
+If a user downgrades their CSI drivers or sidecars, any existing volumes using
+ReadWriteOnce should continue working (switching from `SINGLE_NODE_MULTI_WRITER`
+to `SINGLE_NODE_WRITER`). This behavior is ultimately up to each CSI driver, but
+they should be designed with this backwards compatibility in mind.
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+#### API Server Version N / Scheduler Version N / Kubelet Version N-1 or N-2
+
+When starting two pods with both using the same PVC with ReadWriteOncePod, one pod
+will successfully start, but the other will not be scheduled due to the
+ReadWriteOncePod access mode conflict.
+
+When starting the same two pods but also setting `pod.spec.nodeName` to the same
+node, kubelet will not enforce the access mode and will proceed with starting
+both pods.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+-->
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: ReadWriteOncePod
+  - Components depending on the feature gate:
+    - API server
```

Comment:
Also scheduler, kube-controller-manager, and kubelet

Reply:
Done